doc/Language/syntax.pod6

```diff
@@ -223,109 +223,193 @@ say 'code again';
 
 =head2 X<Identifiers|identifier,identifiers>
 
-Identifiers are grammatical building blocks that occur in several kind
-of statements, giving a name to objects, data structures and blocks of
-code. These names must start with an alphabetic character (or an
-underscore), followed by zero or more word characters (alphabetic,
-underscore or number). You can also embed dashes C<-> or single quotes
-C<'> in the middle, but not two in a row, and only if followed
-immediately by an alphabetic character.
+Identifiers are grammatical building blocks that may be used to give a name
+to entities/objects such as constants, variables (e.g. Scalars) and routines
+(e.g. Subs and Methods). In a variable name, an identifier is normally preceded
+by a sigil; i.e. the sigil is not part of the identifier, only of the variable
+name.
+
+    constant c = 299792458;         # Identifier "c" names an Int
+    my $a = 123;                    # Identifier "a" in the name "$a" of a Scalar
+    sub hello { say "Hello!" };     # Identifier "hello" names a Sub
+
+Identifiers come in different forms: ordinary identifiers, extended
+identifiers, and compound identifiers.
+
+=head3 Ordinary identifiers
+
+An ordinary identifier is composed of a leading alphabetic character
+that may be followed by one or more alphanumeric characters. It may also
+contain isolated, embedded apostrophes C<'> and/or hyphens C<->, provided
+that the next character is each time alphabetic.
+
+The definitions of "alphabetic" and "alphanumeric" include appropriate Unicode
+characters. Which characters are "appropriate" depends on the implementation.
+In the Rakudo/MoarVM Perl 6 implementation alphabetic characters include
+Unicode characters in the character category I<Letter>, and the underscore
+C<_>. Alphanumeric characters additionally include Unicode characters in the
+general category I<Number, Decimal Digit> (Nd).
 
```

Comment:
Maybe just saying above that _ is considered alphanumeric is enough. That is not a general consideration, however. It's generally considered punctuation https://stackoverflow.com/questions/8509262/will-alphanumeric-contain-and-space

Reply:
It is my understanding that in Perl6 the underscore is considered "alphabetic", and thus also "alphanumeric", despite its Unicode classification al punctuation. Hence its special mention in the text above.

 say '_' ~~ /<alpha>/
｢_｣
 alpha => ｢_｣

The rest of the text clarifies precisely what Unicode code points are acceptable, which I'd say is useful information:
https://stackoverflow.com/questions/34689850/whats-allowed-in-a-perl-6-identifier

Reply:
Yes it is, and it's not general, that's why just saying that _ is considered alphanumeric in Perl 6 is enough.

Reply:
I don't understand what you are trying to say. In all honesty, I have that problem with some other comments of yours as well. I appreciate that you take the time to review the PR, but I think we could move faster if you'd be a bit more explicit about what you have in mind.

I'm currently guessing that the only thing you want the text to say it that the underscore is alphanumeric (which I think should be: alphabetic), and hence that the other information should be deleted because it is not "general" in the sense that in does not relate to the Perl6 language, but to an implementation thereof. Is that correct?

If so: I would disagree. This information is pure gold. I am a newbie to Perl6 but have experience with other programming languages (other than Perl). Never have I experienced such a high barrier to entry as with Perl6. This is largely because clear, practical information is oftentimes difficult to find. A question that I think should be answered in every Perl6 101 is which characters are acceptable as part of an identifier. Yet, I have found no documentation that documents this in complete detail. Not in the current docs, not in the synopses, not in Think Perl 6 (Rosenfeld), not in Perl6 Fundamentals (Lenz) and not in Learning Perl6 (Foy). I'd say that's ridiculous. If I look up the Python documentation, I find this: https://docs.python.org/3/reference/lexical_analysis.html#identifiers. Holy Moly - what do you know: a clear answer! 

Why withhold this information, especially having regard to the fact that one of Perl6' flagship features is all-round Unicode support?

Reply:
I'm sorry. I try to, but sometimes it's difficult to say precisely what I mean. Please consider them suggestions, and at the end of the day, none of them will break anything; if something is LTA, we can eventually fix it when merged. But let's get to the point. I kind of try and make the point that anything that is considered alphanumeric *in Perl 6* is OK as an indentifier. Plus `_` is also considered alphanumeric in Perl 6 (not elsewhere). The best way to express that is up to you; maybe the first sentence is already OK in the text, and then a note about `_`.

Reply:
Well, everything that is considered alphanumeric in Perl6 may be OK as an identifier, but not for every position in the identifier. For instance, as a(n unquoted) part of the proposed text states, an ordinary identifier should start with an 'alphabetic' character (which may be an underscore, because the underscore is considered alphabetic, not just alphanumeric). This is why it is important to make a distinction between alphabetic and alphanumeric characters, and why I think that the text as currently proposed is concise and precise (to the best of my knowledge anyway).

Reply:
That's correct, but it does not belong here, where you're only talking about _ as alphanumeric.

Reply:
Anyway, if you think it's OK I'll just resolve this.

Reply:
I'm sorry; I'm trying hard to see where the flaw is in the text, or how you are reading it so the flaw comes out, but I don't see it. In particular the text never explicitly states that the underscore _ is alphanumeric. It first says it is *alphabetic*, and then implies - through the phrase "Alphanumeric characters *additionally* include..." that is is also alphanumeric. But I need to know it is alphabetic too, otherwise I don't know I can use it in certain positions in identifiers.

Perhaps it's better to resolve this conversation.

Reply:
:+1: